pallets/manta-pay/src/errors.rs

```diff
@@ -0,0 +1,349 @@
+            TransferPostError::<T>::DuplicateSpend => Self::DuplicateSpend,
+            TransferPostError::<T>::InvalidProof => Self::InvalidProof,
+            TransferPostError::<T>::UnexpectedError(e) => match e {
+                TransferLedgerError::ChecksumError => Self::TransferLedgerChecksumError,
+                TransferLedgerError::VerifyingContextDecodeError(_) => {
+                    Self::TransferLedgerVerifyingContextDecodeError
+                }
+                TransferLedgerError::FpEncodeError(_) => Self::TransferLedgerFpEncodeError,
+                TransferLedgerError::FungibleLedgerError(err) => err.into(),
+                TransferLedgerError::UnknownAsset => Self::TransferLedgerUnknownAsset,
+                TransferLedgerError::InvalidTransferShape => Self::InvalidShape,
+                TransferLedgerError::ProofSystemError(_) => Self::TransferLedgerProofSystemFailed,
+                TransferLedgerError::InvalidProof => Self::InvalidProof,
+                TransferLedgerError::Marker(_) => Self::Marker,
+                TransferLedgerError::SenderLedgerError(err) => SenderPostError::from(err).into(),
+                TransferLedgerError::InvalidAssetId => Self::InvalidAssetId,
+                TransferLedgerError::ReceiverLedgerError(err) => {
+                    ReceiverPostError::from(err).into()
+                }
+            },
+        }
+    }
+}
+
+/// Transfer Ledger Error
+pub enum TransferLedgerError<T>
+where
+    T: Config,
+{
+    /// Wrong Checksum Error
+    ChecksumError,
+
+    /// Verifying Context Decoding Error
+    VerifyingContextDecodeError(VerifyingContextError),
+
+    /// Field Element Encoding Error
+    FpEncodeError(scale_codec::Error),
+
+    /// Unknown Asset Error
+    UnknownAsset,
+
+    /// Fungible Ledger Error
+    FungibleLedgerError(FungibleLedgerError),
+
+    /// Sender Ledger Error
+    SenderLedgerError(SenderLedgerError),
+
+    /// Receiver Ledger Error
+    ReceiverLedgerError(ReceiverLedgerError<T>),
+
+    /// Invalid Transfer Shape
+    InvalidTransferShape,
+
+    /// Proof System Error
+    ProofSystemError(ProofSystemError<config::Config>),
+
+    /// Invalid Transfer Proof Error
+    ///
+    /// Validity of the transfer could not be proved by the ledger.
+    InvalidProof,
+
+    /// Invalid AssetId, cannot have a value of zero
+    InvalidAssetId,
+
+    /// Type Marker Parameter
+    Marker(PhantomData<T>),
+}
+
+impl<T> From<TransferLedgerError<T>> for TransferPostError<T>
+where
+    T: Config,
+{
+    #[inline]
+    fn from(value: TransferLedgerError<T>) -> Self {
+        match value {
+            TransferLedgerError::InvalidProof => Self::InvalidProof,
+            TransferLedgerError::InvalidTransferShape => Self::InvalidShape,
+            TransferLedgerError::SenderLedgerError(err) => Self::Sender(err.into()),
+            TransferLedgerError::ReceiverLedgerError(err) => Self::Receiver(err.into()),
+            err => Self::UnexpectedError(err),
```

Comment:
I thought we wanted to get rid of or at least minimize the unexpected/internal-ledger-error
This catch all still won't describe/trace the FungibleLedgerError -> TransferLedgerError -> TransferPostError -> PalletError transform.
Maybe we should also add a debug log ?

Reply:
Actually this `UnexpectedError` is from manta-rs there isn't really a way to remove it without altering manta-rs. 

It gets resolved in an exhaustive pattern to the pallet error type here.
```rust 
impl<T> From<TransferPostError<T>> for Error<T>
```

I agree it is a bit convoluted but functionally it should point us to the exact error